src/ff1.rs

```diff
@@ -146,6 +146,95 @@ impl RadixOps for PowerTwoRadix {
     }
 }
 
+/// A numeral string with radix 2.
+pub struct BinaryNumeralString(Vec<u8>);
+
+impl BinaryNumeralString {
+    /// Creates a BinaryNumeralString from a byte slice, with each byte
+    /// interpreted in little-endian bit order.
+    pub fn from_bytes_le(s: &[u8]) -> Self {
+        let mut data = Vec::with_capacity(s.len() * 8);
+        for n in s {
+            let mut tmp = *n;
+            for _ in 0..8 {
+                data.push(tmp & 1);
+                tmp >>= 1;
+            }
+        }
+        BinaryNumeralString(data)
+    }
+
+    /// Returns a Vec<u8>, with each byte written from the BinaryNumeralString
+    /// in little-endian bit order.
+    pub fn to_bytes_le(&self) -> Vec<u8> {
+        let mut data = Vec::with_capacity((self.0.len() + 7) / 8);
+        let mut acc = 0;
+        let mut shift = 0;
+        for n in &self.0 {
+            acc += n << shift;
+            shift += 1;
+            if shift == 8 {
+                data.push(acc);
```

Comment:
Also, if the length of self.0 were not divisible by 8, would we sometimes be dropping bits on the floor?

Reply:
Ooh yes, good catch - for this to be correct in the general case, we should push whatever remains in `acc` once we're done iterating over `self.0`. But per above, we will always have a multiple of 8 bits, so this is fine.